### ISSUE TITLE
Publication to Gradle Plugin Portal

### ISSUE THREAD

**Issue by Vampire** [ajoberstar/grgit#359]
More a question but a bug, but I thought I ask here so that other will maybe also see the question and answer here.
I'm curious, why don't you publish to Gradle Plugin Portal anymore?
Yes, with it redirecting to JCenter for things it does not have
and with JCenter forwarding to Maven Central for things it does not have it still works and "only" discoverability is affected.

But, the "only" is in quotes, because that also means if for example JCenter has one of its regular outtakes, it does not work.
Or if the connection from JCenter to Maven Central has a problem, it does not work.

And imho even more importantly, the discoverability is not really an "only".
If I'm in need for a Gradle plugin I almost always exclusively ask the Plugin Portal, as it is the canonical source for Gradle plugins.
It is very seldom that I use Google for that, even if I didn't find something.

And while your plugin is of course still listed with the old versions, I for example also have an eye on when the last version of a plugin was published and so on.

So may I ask what your reason is / reasons are not to publish to the Plugin Portal anymore?
Especially when it was already set up and working properly. 5.0.0-rc.3 was still published to it.

**Comment by ajoberstar** [comment]
TL;DR Too much publishing churn; I'm done with it. I'm just using Central now. The discoverability isn't important enough to me.

Apologies for this being a flat-out rant... 

I've been maintaining this plugin for 10 years now, so I've been through all of the Gradle/community direction changes on this (can't remember the order of all of them):
- Started with publishing to Central via the `maven` plugin
- The switch from `maven` to `maven-publish` (when `maven-publish` didn't work with signing, meaning it was useless for publishing to central)
- Switched to Bintray with the old attribute based portal, with the flaky `com.jfrog.bintray`
- Switched to the `com.gradle.plugin-publish` portal approach while maintaining JCenter publishing
- Switched to `maven-publish` for publishing to Bintray after difficulties dealing with hacky afterEvaluate and similar behavior in the bintray plugin
- Adopted nebula dependency locking and their approach to ensure static versions in the POM
- Change locking again as Gradle made it available and eventually supported a static version approach
- Ongoing mystery along the way of why the `com.gradle.plugin-publish` ignores config I'm doing with `maven-publish`
- JCenter announces it's shutting down, causing the extract and blurb you reference in #357
- I ignored the plugin for a year out of exasperation with all of this, only to have accidentally published dynamic versions in the POM, which eventually caused people's builds to break when JGit 6 came out. Resolving that issue forced me back to Maven Central.
- JCenter continues to exist in zombie state anyway, with nearly no visibility (all I can find is a update note on the top of their original shutdown announcement)

When `com.gradle.plugin-publish` was first created I had established plugins with existing maven coordinates. Back then the doc said, declare those coordinates explicitly to avoid a `gradle.plugin.whatever` namespace. Nowadays, the plugin says to remove `mavenCoordinates` if you want to use the POM created by `maven-publish`. Nowhere in that time was the `com.gradle.plugin-publish` plugin documented at anywhere near the quality of Gradle's open source components. And never were its release notes easily discoverable and consumable. It's also been 0.x version the entire time, even though it's literally the only supported way to publish to the plugin. And never as far as I can tell was the plugin open source, so that I could just figure out for myself how it worked.

The portal itself does let you search for plugins, but only does a basic keyword search and sorts by most recently published. You can't get any hints of what plugins are commonly used, you can't see any documentation or anything else about the plugin besides a link back to the project's home. It's nearly useless besides being a quick way to look up the latest version of a plugin you already know exists.

Some of this is self-inflicted early adopter syndrome, but I just don't want to think about publishing anymore.

**Comment by Vampire** [comment]
Thanks for the comprehensive answer.
